database/migrations/2018_01_26_211937_make_bucket_and_details_columns_nullable.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class MakeBucketAndDetailsColumnsNullable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('posts', function (Blueprint $table) {
+            $table->string('action_bucket')->nullable()->change();
```

Comment:
I see why `details` would be `null`, but why would `action_bucket` be `null` for anything new?

Reply:
Yeah, I'm curious if we keep this non-nullable but have it default to `default` (or w/e)? Otherwise I could see bugs popping up since we'd have to always keep in mind both possible "states" of this column.

```
$table->string('action_bucket')->default('default')->change();
```

Reply:
To support old tests, old posts, or old requests that might not be sending it. I mean we can probably deal with support for `action_bucket` in more detail. But it is odd that old posts now have an empty string in that column since it was before "buckets" were a thing. 

If we don't want to allow this to be `nullable` then we probably need a sensible default value.

Reply:
Ok, i just remember running into issues storing `default` everywhere before (for images). But if we are good with that then I can change it.